tests/foreman/sys/test_katello_certs_check.py

```diff
@@ -41,33 +42,40 @@
 class KatelloCertsCheckTestCase(TestCase):
     """Implements ``katello-certs-check`` tests.
 
-    Depends on presence of custom certificates at path given
-    in robottello.properties file.
+    Depends on presence of scripts and files in
+    tests/foreman/data/
+    CA cert (a.k.a cacert.crt or rootCA.pem) can be used as bundle file.
     """
 
     @classmethod
-    @skip_if_not_set('certs')
     def setUpClass(cls):
-        """Get host name and credentials."""
+        """Get host name, scripts, and create working directory."""
         super(KatelloCertsCheckTestCase, cls).setUpClass()
-        _, cls.ca_bundle_file_name = os.path.split(settings.certs.ca_bundle_file)
-        _, cls.cert_file_name = os.path.split(settings.certs.cert_file)
-        _, cls.key_file_name = os.path.split(settings.certs.key_file)
-        cls.ca_bundle_file = settings.certs.ca_bundle_file
-        cls.cert_file = settings.certs.cert_file
-        cls.key_file = settings.certs.key_file
+        _, cls.sat6_hostname = os.path.split(settings.server.hostname)
+        cls.key_file_name = '{0}/{0}.key'.format(cls.sat6_hostname)
+        cls.cert_file_name = '{0}/{0}.crt'.format(cls.sat6_hostname)
+        cls.ca_bundle_file_name = 'cacert.crt'
         cls.SUCCESS_MSG = "Validation succeeded"
-        # uploads certs to satellite
-        upload_file(
-            local_file=settings.certs.ca_bundle_file,
-            remote_file="/tmp/{0}".format(cls.ca_bundle_file_name),
-        )
         upload_file(
-            local_file=settings.certs.cert_file, remote_file="/tmp/{0}".format(cls.cert_file_name)
+            local_file=get_data_file('generate-ca.sh'), remote_file="generate-ca.sh",
         )
         upload_file(
-            local_file=settings.certs.key_file, remote_file="/tmp/{0}".format(cls.key_file_name)
+            local_file=get_data_file('generate-crt.sh'), remote_file="generate-crt.sh",
         )
+        upload_file(local_file=get_data_file('openssl.cnf'), remote_file="openssl.cnf")
+        upload_file(local_file=get_data_file('serial'), remote_file="serial")
+        # create the CA cert.
+        with get_connection(timeout=300) as connection:
+            result = connection.run("chmod +x generate-ca.sh")
+            result = connection.run("bash generate-ca.sh")
```

Comment:
I'm pretty sure you don't need executable bit when you pass file as argument to interpreter. I.e. `chmod` can be safely dropped.

Reply:
true

Reply:
improvements made, but testing still pending